android/app/src/main/java/com/epfl/dedis/hbt/ui/wallet/RxAmountFragment.kt

```diff
@@ -39,12 +40,28 @@ class RxAmountFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val receiveButton = binding.walletButtonReceive
+
         val amount = binding.walletRxAmount
+        val datetime = System.currentTimeMillis()
+
+        binding.walletButtonReceive.setOnClickListener {
+            walletViewModel.transitionTo(ReceiverShow(amount.text.toString().toFloat(), datetime))
```

Comment:
I would write `ReceiverShow(amount, datetime)` for readability, and let `ReceiverShow` do the translation `amount.text.toString().toFloat()`

Reply:
Applied